Lombiq.Tests.UI/Services/Counters/Middlewares/PageLoadProbeMiddleware.cs

```diff
@@ -0,0 +1,24 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using System;
+using System.Threading.Tasks;
+
+namespace Lombiq.Tests.UI.Services.Counters.Middlewares;
+
+public class PageLoadProbeMiddleware
```

Comment:
Why is this and `NavigationProbe` both necessary?

Reply:
The `NavigationProbe` is for the navigation requested by the test code e.g. `UITestContext.GoToHomePageAsync(..)` while the `PageLoadProbe` is for the requests from the web browser e.g. image requests, API calls, etc. The `NavigationProbe` captures image requests while the page loading, but not async calls.

Reply:
Wouldn't `PageLoadProbe` alone enough? Since all navigations requested by the test will in the end result in a request.

BTW I'd rather call this `RequestLoadProbleMiddleware` then.